src/operator/controllers/otterizeclient/client.go

```diff
@@ -69,7 +69,21 @@ func (c *CloudClient) RegisterK8SPod(ctx context.Context, namespace string, _ st
 	return res.RegisterKubernetesPodOwnerCertificateRequest.Id, nil
 }
 
-func (c *CloudClient) CleanupOrphanK8SPodEntries(_ context.Context, _ string, _ map[string]*goset.Set[string]) error {
+func (c *CloudClient) CleanupOrphanK8SPodEntries(ctx context.Context, _ string, existingServicesByNamespace map[string]*goset.Set[string]) error {
```

Comment:
From the name and the API, it really isn't clear how this works. What this does is report all the pods that exist _right now_ - it isn't clear how this does clean up.

Maybe the API - and the function that implements it, should be called something like, ReportNamespacePodOwnersForCertificateRequestCleanup?

Basically, anything that says: "this reports all of the pod owners, and it is used for cleanup"

Right now, it says: removeOrphanedCertificateRequests, and sends all of the pod owners. This seems like it should delete all of them.